kube/lib/wire.libsonnet

```diff
@@ -0,0 +1,181 @@
+local kube = import "kube.libsonnet";
+
+// Wire deployment.
+
+{
+    local wire = self,
+
+    // A Wire Environment. Also called a Deployment.
+    // Contains all services/components required to run Wire on k8s, apart from
+    // per-cluster services.
+    // All components will be run in a given k8s namespace, by default named
+    // after the environment.
+    Environment(name):: {
+        local env = self,
+        local cfg = env.cfg,
+
+        // User configurable.
+        cfg:: {
+            // Name of environment, used internally. Can be overridden.
+            name: name,
+            // Kubernetes namespace in which environment will be run. Can be overridden.
+            namespace: name,
+
+            // Public DNS names to be used by wire-server.
+            dns: {
+                // Top-level domain. Must be set, or all specific domain names
+                // underneath must be overriden.
+                domain: error "dns.domain must be set", // .my-wire.example.com
+
+                // Default DNS prefixes for top-level domain.
+                nginzHTTPS: "nginz-https.%s" % [cfg.dns.domain],
+                nginzTLS: "nginz-tls.%s" % [cfg.dns.domain],
```

Comment:
You decided to diverge from `nginz-ssl` used in other places :) 

I suppose we should find a naming consensus that we're all happy with.